### DISCUSSION TITLE
Combination of text and form fields in RTE

### DISCUSSION THREAD

**Discussion by shanthi-techie** [react-page/react-page#1116] [Q&A] [answered]
Hi All,

I've a use case to combine text and form fields in the same cell plugin i can say. Is it possible to combine both text and form fields like input text box, check box , radio button. As per the demo its not possible. what i'm trying to achieve is something like the below.

![image](https://user-images.githubusercontent.com/66308775/146558240-c945d52c-7597-4f7a-92c8-02e1eb643d07.png)

**Comment by macrozone** [comment]
this is an interessing one!

I assume that the user (the editor) that writes the text writes some kind of template and end-users then can fill it out?

I'll preparing something for that...

**Comment by macrozone** [comment]
hi @shanthi-techie in the latest 4.4.0 version (beta) you can do it easily. here is an example:

https://react-page.github.io/beta/examples/formFieldInText

see the code https://github.com/react-page/react-page/blob/beta/examples/pages/examples/formFieldInText.tsx


https://user-images.githubusercontent.com/1972353/146642320-4de9427f-8e0f-4701-a974-38c11c47a967.mov

**Comment by shanthi-techie** [comment, marked as answer]
Awesome. thanks for the quick fix. Can this be implemented only in the beta version? The documentation says the beta version has unstable features. Is it ok to use the beta version for project development?

**Reply by macrozone** [reply]
I released it to beta just because there are some changes that i did not test yet in certain scenarios  and also for you to test it and give feedback.

apart from this change, its the same as the current stable version. if everything is alright, i'll release it to stable soon.

**Reply by shanthi-techie** [reply]
https://github.com/react-page/react-page/blob/beta/examples/pages/examples/formFieldInText.tsx . The above link is not working in my local with react-page@beta 4.4.0-1 version. The project is building successfully only with 4.3.1 version of react-page

**Reply by macrozone** [reply]
sorry i don't understand your problem. This link is an url that points to github.

**Reply by shanthi-techie** [reply]
I'm trying to run the project in my local with  react-page : beta version. (4.4.0) but the build gets failed.  I'm able ot build this project only with 4.3.1 version. I'm trying to run it with 4.4.0 so that i can test formFieldInText feature but everytime i get error on cellPlugins when i upgrade it to 4.40

**Reply by macrozone** [reply]
@shanthi-techie can you show a reproduction? (code examples, exact versions, logs, error messages, etc.)

**Reply by shanthi-techie** [reply]
package.json: react-page version is beta.1
![image](https://user-images.githubusercontent.com/66308775/146775586-a5b7d27d-dd97-44a9-8d84-f3131783b5f6.png)

Once after i've updated this to beta version i'm getting the below error,

![image](https://user-images.githubusercontent.com/66308775/146775857-e72ffc2b-7d15-423e-bd4f-0be61fad08ab.png)

**Reply by shanthi-techie** [reply]
Its the same code from beta feature so i did not share code base with you

**Reply by macrozone** [reply]
i see, thank you

**Reply by macrozone** [reply]
is this your own code base? or did you pull the repo? the versions of your different @react-page plugins do not match. in the repo it should say 0.0.0 as it is a lerna monorepo. In your own code base you should use the exact same versions (4.4.0-beta.1).

also currently there is a problem with typescript strict mode, so you have to turn that off until https://github.com/react-page/react-page/issues/1104 is fixed

**Reply by shanthi-techie** [reply]
yup, it works perfectly. Thanks a lot.

**Reply by shanthi-techie** [reply]
@macrozone This feature is working fine with 4.4.0-beta.1 but not with 4.3.1 version. When this will be released in 4.3.1 ?

**Reply by macrozone** [reply]
@shanthi-techie i released it now to production, its available with version 4.4.0
